test/conftest.py

```diff
@@ -0,0 +1,16 @@
+import os
+
+
+def pytest_sessionstart(session):
+    cwd = os.fspath(os.getcwd())
+    if cwd.endswith("image_and_tests"):
+        os.chdir("test")
```

Comment:
Some commentary on this:

My current version of this for conftest.py is as follows:
```python
def pytest_sessionstart(session):
    os.chdir(Path(__file__).parent)
    sys.path.insert(0, str(Path(__file__).parents[1]))```

I've modified it to use your `__file__` logic which works well, but without this block, there are a few undesirable effects on the testing suite as I intended to design it:
- Adding the git repo root directory to sys.path (or doing something similar) is necessary for Python to see the docker_cli module if it isn't installed and pytest isn't being run from that directory.
- Moving to the test directory enables the test suite to test in a controlled environment that acts how a user might use the code, including pulling in a dockerfile from the local file - if I don't do this, then I have to add the full path to the dockerfile every time, which leaves some of what is written in Image.py untested.
- I have also moved the `test_image_id` fixture to `conftest.py` which makes it available for use in other testing files

I do plan to get rid of the `pytest_sessionfinish` function and instead use temporary files, but in particular pytest_sessionstart seems to be the best single place to move the directory and get Python to recognize `docker_cli` from anywhere in the system. Any thoughts?

Reply:
>I have also moved the `test_image_id` fixture to `conftest.py` which makes it available for use in other testing files

Instead of this, how about creating a `test/utils.py` (or something similarly named) file and putting `test_image_id` in there?

That way, you'll need to explicitly import it in your testing scripts using

```python
from .utils import test_image_id
```

which seems more intuitive and pythonic, instead of it magically becoming available without being imported.

Note that you'll need to add an (empty) `__init__.py` file to your `test` directory in order to make the import work.